pallets/services/src/lib.rs

```diff
@@ -0,0 +1,721 @@
+			request_id: u64,
+			/// The ID of the service blueprint.
+			blueprint_id: u64,
+		},
+
+		/// A service request has been updated or modified.
+		ServiceRequestUpdated {
+			/// The account that requested the service.
+			owner: T::AccountId,
+			/// The ID of the service request.
+			request_id: u64,
+			/// The ID of the service blueprint.
+			blueprint_id: u64,
+			/// The list of service providers that need to approve the service.
+			pending_approvals: Vec<T::AccountId>,
+			/// The list of service providers that atomaticaly approved the service.
+			approved: Vec<T::AccountId>,
+		},
+		/// A service has been initiated.
+		ServiceInitiated {
+			/// The owner of the service.
+			owner: T::AccountId,
+			/// The ID of the service request that got approved (if required).
+			request_id: Option<u64>,
+			/// The ID of the service.
+			service_id: u64,
+			/// The ID of the service blueprint.
+			blueprint_id: u64,
+		},
+
+		/// A service has been terminated.
+		ServiceTerminated {
+			/// The owner of the service.
+			owner: T::AccountId,
+			/// The ID of the service.
+			service_id: u64,
+			/// The ID of the service blueprint.
+			blueprint_id: u64,
+		},
+
+		/// A job has been called.
+		JobCalled {
+			/// The account that called the job.
+			caller: T::AccountId,
+			/// The ID of the service.
+			service_id: u64,
+			/// The ID of the call.
+			call_id: u64,
+			/// The index of the job.
+			job: u8,
+			/// The arguments of the job.
+			args: Vec<Field<T::AccountId>>,
+		},
+
+		/// A job result has been submitted.
+		JobResultSubmitted {
+			/// The account that submitted the job result.
+			provider: T::AccountId,
+			/// The ID of the service.
+			service_id: u64,
+			/// The ID of the call.
+			call_id: u64,
+			/// The index of the job.
+			job: u8,
+			/// The result of the job.
+			result: Vec<Field<T::AccountId>>,
+		},
+	}
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	// Counters
+
+	/// The next free ID for a service blueprint.
+	#[pallet::storage]
+	#[pallet::getter(fn next_blueprint_id)]
+	pub type NextBlueprintId<T> = StorageValue<_, u64, ValueQuery>;
+
+	/// The next free ID for a service request.
+	#[pallet::storage]
+	#[pallet::getter(fn next_service_request_id)]
+	pub type NextServiceRequestId<T> = StorageValue<_, u64, ValueQuery>;
+
+	/// The next free ID for a service Instance.
+	#[pallet::storage]
+	#[pallet::getter(fn next_instance_id)]
+	pub type NextInstanceId<T> = StorageValue<_, u64, ValueQuery>;
+
+	/// The next free ID for a service call.
+	#[pallet::storage]
+	#[pallet::getter(fn next_job_call_id)]
+	pub type NextJobCallId<T> = StorageValue<_, u64, ValueQuery>;
+
+	/// The service blueprints along with their owner.
+	#[pallet::storage]
+	#[pallet::getter(fn blueprints)]
+	pub type Blueprints<T: Config> = StorageMap<
+		_,
+		Identity,
+		u64,
+		(T::AccountId, ServiceBlueprint),
+		ResultQuery<Error<T>::BlueprintNotFound>,
+	>;
+
+	/// The service providers for a specific service blueprint.
```

Comment:
Can we call this Operators? That's how it'll be communicated in the docs/system.